controllers/oscmachine_controller.go

```diff
@@ -355,14 +391,32 @@ func (r *OscMachineReconciler) reconcileDelete(ctx context.Context, machineScope
 			return reconcileDeleteVolume, err
 		}
 	}
+
+	ctx, _, publicIpDone := tele.StartSpanWithLogger(ctx, "controllers.OscMachineReconciler.reconcilePublicIpDelete")
+	defer publicIpDone()
```

Comment:
Wouldn't it be more useful to call publicIpDone after the call of the functions ? Because here we we defer at the end of the function